sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileSystemClient.cs

```diff
@@ -502,6 +502,154 @@ private void SetNameFieldsIfNull()
+                        LastModified = containerResponse.Value.LastModified
+                    },
+                    containerResponse.GetRawResponse());
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+            finally
+            {
+                scope.Dispose();
+            }
+        }
+
+        /// <summary>
+        /// The <see cref="CreateIfNotExistsAsync"/> operation creates a new file system
+        /// under the specified account. If the file system with the same name
+        /// already exists, the operation fails.
+        ///
+        /// For more information, see <see href="https://docs.microsoft.com/rest/api/storageservices/create-container"/>.
+        /// </summary>
+        /// <param name="publicAccessType">
+        /// Optionally specifies whether data in the file system may be accessed
+        /// publicly and the level of access. <see cref="Models.PublicAccessType.FileSystem"/>
+        /// specifies full public read access for file system and path data.
+        /// Clients can enumerate paths within the file system via anonymous
+        /// request, but cannot enumerate file system within the storage
+        /// account.  <see cref="Models.PublicAccessType.Path"/> specifies public
+        /// read access for pathss.  Path data within this file system can be
+        /// read via anonymous request, but file system data is not available.
+        /// Clients cannot enumerate pathss within the file system via anonymous
+        /// request.  <see cref="Models.PublicAccessType.None"/> specifies that the
+        /// file system data is private to the account owner.
+        /// </param>
+        /// <param name="metadata">
+        /// Optional custom metadata to set for this file system.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{ContainerInfo}"/> describing the newly
+        /// created container.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual async Task<Response<FileSystemInfo>> CreateIfNotExistsAsync(
+            Models.PublicAccessType publicAccessType = Models.PublicAccessType.None,
+            Metadata metadata = default,
+            CancellationToken cancellationToken = default)
+        {
+            DiagnosticScope scope = ClientDiagnostics.CreateScope($"{nameof(DataLakeFileSystemClient)}.{nameof(CreateIfNotExists)}");
+            try
+            {
+                scope.Start();
+                Response<BlobContainerInfo> containerResponse = await _containerClient.CreateIfNotExistsAsync(
```

Comment:
Nit - this is getting close to enough logic that it might be nice to share a sync/async implementation with a `CreateIfNotExistsInternal(..., bool async)`.

Reply:
```DataLakeFileSystemClient.CreateIfNotExists()``` and ```DataLakeFileSystemClient.CreateIfNotExistsAsync()``` call ```BlobContainerClient.CreateIfNotExists()``` and ```BlobContainerClient.CreateIfNotExistsAsync()```, respectively.  ```BlobContainerClient.CreateIfNotExistsInternal()``` is private.  I think creating a shared ```DataLakeFileSystemClient.CreateIfNotExistsInternal()``` would require duplicated most of the logic in ```BlobContainerClient.CreateIfNotExistsInternal()```, so it would be a wash in terms of duplicate code.

Reply:
I think it could just be the following in a shared method without exposing the internal blob method:

```C#
Response<BlobContainerInfo> containerResponse = async ?
    await _containerClient.CreateIfNotExistsAsync(...).ConfigureAwait(false) :
    _containerClient.CreateIfNotExists(...);
```

Again though, it doesn't matter as much for this one but it would be nice if/when you're adding other helpers with more complicated logic.

Reply:
I'll add for ```DataLakeFileSystemClient```.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/10028